src/itemstackmetadata.h

```diff
@@ -21,15 +21,37 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define ITEMSTACKMETADATA_HEADER
 
 #include "metadata.h"
+#include "tool.h"
 
 class Inventory;
 class IItemDefManager;
 
 class ItemStackMetadata : public Metadata
 {
 public:
+	ItemStackMetadata() : toolcaps_overridden(false) {}
+
+	// Overriddes
+	void clear();
+	bool setString(const std::string &name, const std::string &var);
```

Comment:
use the override keyword, eg:  `void clear() override;`

Reply:
Doesn't this only work for virtual methods?

Reply:
Oh, they are virtual.